### ISSUE TITLE
Do not allow disabling notification in Oreo (IllegalStateException: Not allowed to start service Intent)

### ISSUE THREAD

**Issue by mwitkow** [julian-klode/dns66#265]
DNS66 fails periodically and requires a manual restart now and then. This seems like the exception 

```
--------- beginning of crash
04-05 14:43:10.045 10695 10695 E AndroidRuntime: FATAL EXCEPTION: main
04-05 14:43:10.045 10695 10695 E AndroidRuntime: Process: org.jak_linux.dns66, PID: 10695
04-05 14:43:10.045 10695 10695 E AndroidRuntime: java.lang.RuntimeException: Unable to start receiver org.jak_linux.dns66.vpn.BootComplete: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.jak_linux.dns66/.vpn.AdVpnService (has extras) }: app is in background uid UidRecord{8a4e8f9 u0a164 RCVR idle procs:1 seq(0,0,0)}
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:3424)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at android.app.ActivityThread.-wrap17(Unknown Source:0)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1772)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:105)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6809)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.jak_linux.dns66/.vpn.AdVpnService (has extras) }: app is in background uid UidRecord{8a4e8f9 u0a164 RCVR idle procs:1 seq(0,0,0)}
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1522)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at android.app.ContextImpl.startService(ContextImpl.java:1478)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at android.content.ContextWrapper.startService(ContextWrapper.java:645)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at android.content.ContextWrapper.startService(ContextWrapper.java:645)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at org.jak_linux.dns66.vpn.AdVpnService.checkStartVpnOnBoot(AdVpnService.java:146)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at org.jak_linux.dns66.vpn.BootComplete.onReceive(BootComplete.java:23)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:3410)
04-05 14:43:10.045 10695 10695 E AndroidRuntime: 	... 8 more
-----
```

Phone One Plus 5t
Oreo 8.0.0
Oxygen OS 5.0.4

**Comment by mwitkow** [comment]
Enabling the notification and then muting it in the Oreo notification system works. 

However I think it's still worth fixing to make sure that this gotcha is not an issue for other users, eg by removing the notification menu option on Oreo and having it on always.

**Comment by soenkegissel** [comment]
I see this exception every other day. It definitely is worth fixing.

**Comment by julian-klode** [comment]
This was fixed in 2e771b2128dbb44b8e56190d47840cfea382292d back in February
